Segment trees/rangeSumQuery.cpp

```diff
@@ -0,0 +1,59 @@
+#include<bits/stdc++.h>
+#define lld long long int
+#define ld long int
+#define str string
+#define dbl double
+using namespace std;
+
+
+void buildSegmentTree(int tree[],int values[],int index,int start,int end){
+	if(start>end) return;
+	if(start==end){
+		tree[index]=values[start];
+	}
+	else{
+		int mid = (start+end)/2;
+
+		//left subTree
+		buildSegmentTree(tree,values,2*index,start,mid);
+		//right subTree
+		buildSegmentTree(tree,values,2*index+1,mid+1,end);
+
+		tree[index] = tree[2*index]+tree[2*index+1];
```

Comment:
For example, in the generic segment tree described in my comment, line 22 would be:
`` tree[index] = f( tree[2*index], tree[2*index+1] );``